addon/mixins/flexberry-maplayer-actions-handler.js

```diff
@@ -364,5 +367,47 @@ export default Ember.Mixin.create({
 
     Ember.set(layer, 'isDeleted', true);
     return layer;
-  }
+  },
+
+  /**
+   Gets options for setIndexes method.
+
+    @method getIndexesOptions
+    @param {Object} layer Layer to get options for.
+    @returns {Object} First index and layers (options.firstIndex and options.layers).
+  */
+  getIndexesOptions(layer) {
+    let options = {};
+    let parent = layer.get('parent');
+    if (Ember.isNone(parent.get('parent'))) {
+      let firstLayer = parent.get('layers').objectAt(0);
+      options.layers = parent.get('layers');
+      options.firstIndex = firstLayer.get('index');
```

Comment:
If hierarchy is empty at the moment, then firstLayer.get('index') will return undefined.
options.firstIndex = firstLayer.get('index') || 0; will solve this problem.